posthog/hogql/database/database.py

```diff
@@ -409,10 +410,23 @@ def define_mappings(warehouse: dict[str, Table], get_table: Callable):
             )
 
         if "person_id" not in warehouse[warehouse_modifier.table_name].fields.keys():
-            warehouse[warehouse_modifier.table_name].fields["person_id"] = ExpressionField(
-                name="person_id",
-                expr=parse_expr(warehouse_modifier.distinct_id_field),
+            events_join = next(
+                (
+                    join
+                    for join in DataWarehouseJoin.objects.filter(team_id=team.pk).exclude(deleted=True)
+                    if join.joining_table_name == "events"
+                ),
+                None,
             )
+            if events_join:
+                warehouse[warehouse_modifier.table_name].fields["person_id"] = FieldTraverser(
+                    chain=[events_join.field_name, "person_id"]
+                )
+            else:
+                warehouse[warehouse_modifier.table_name].fields["person_id"] = ExpressionField(
+                    name="person_id",
+                    expr=parse_expr(warehouse_modifier.distinct_id_field),
```

Comment:
Do we not want to fallback onto using something like `pdi.person_id` instead here? That is the _true_ `person_id` and not `distinct_id`